ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -1521,42 +1521,151 @@ impl<'a> AsmPrinter<'a> {
+    use lazy_static::lazy_static;
+    use regex::{Regex, RegexBuilder};
+    use std::{collections::HashMap, sync::Arc};
     use ykaddr::addr::symbol_to_ptr;
 
+    /// All x64 registers sorted by class. Later we allow users to match all (e.g.) 8 bit registers with `r.8._`.
+    const X64_REGS: [(&str, &str); 4] = [
+        (
+            "8",
+            "al|bl|cl|dl|sil|dil|spl|bpl|r8b|r9b|r10b|r11b|r12b|r13b|r14b|r15b|ah|bh|ch|dh",
+        ),
+        (
+            "16",
+            "ax|bx|cx|dx|si|di|sp|bp|r8w|r9w|r10w|r11w|r12w|r13w|r14w|r15w",
+        ),
+        (
+            "32",
+            "eax|ebx|ecx|edx|esi|edi|esp|ebp|r8d|r9d|r10d|r11d|r12d|r13d|r14d|r15d",
+        ),
+        (
+            "64",
+            "rax|rbx|rcx|rdx|rsi|rdi|rsp|rbp|r8|r9|r10|r11|r12|r13|r14|r15",
+        ),
+    ];
+
+    const X64_RAW_REGS_MAP: [[&str; 5]; 16] = [
+        ["rax", "eax", "ax", "ah", "al"],
+        ["rbx", "ebx", "bx", "bh", "bl"],
+        ["rcx", "ecx", "cx", "ch", "cl"],
+        ["rdx", "edx", "dx", "dh", "dl"],
+        ["rsi", "esi", "si", "", "sil"],
+        ["rdi", "edi", "di", "", "dil"],
+        ["rsp", "esp", "sp", "", "spl"],
+        ["rbp", "ebp", "bp", "", "bpl"],
+        ["r8", "r8d", "r8w", "", "r8b"],
+        ["r9", "r9d", "r9w", "", "r9b"],
+        ["r10", "r10d", "r10w", "", "r10b"],
+        ["r11", "r11d", "r11w", "", "r11b"],
+        ["r12", "r12d", "r12w", "", "r12b"],
+        ["r13", "r13d", "r13w", "", "r13b"],
+        ["r14", "r14d", "r14w", "", "r14b"],
+        ["r15", "r15d", "r15w", "", "r15b"],
+    ];
+
+    lazy_static! {
+        static ref X64_REGS_MAP: HashMap<&'static str, usize> = {
+            let mut map = HashMap::new();
+            for (i, regs) in X64_RAW_REGS_MAP.iter().enumerate() {
+                for r in regs.iter() {
+                    if *r != "" {
+                        map.insert(*r, i);
+                    }
+                }
+            }
+            map
+        };
+    }
+
+    #[test]
+    fn check_matching() {
+        let fmm = fmatcher("r.8.x r.8.y r.64.x");
+        assert!(fmm.matches("al bl rax").is_ok());
+        assert!(fmm.matches("al al rax").is_err());
+        assert!(fmm.matches("al bl rbx").is_err());
+        assert!(fmm.matches("al bl eax").is_err());
+    }
+
     fn test_module() -> jit_ir::Module {
         jit_ir::Module::new_testing()
     }
 
-    /// Test helper to use `fm` to match a disassembled trace.
-    fn match_asm(cgo: Arc<X64CompiledTrace>, pattern: &str) {
-        let dis = cgo.disassemble().unwrap();
-
+    fn fmatcher<'a>(ptn: &'a str) -> FMatcher<'a> {
         // Use `{{name}}` to match non-literal strings in tests.
         let ptn_re = Regex::new(r"\{\{.+?\}\}").unwrap();
         let ptn_re_ignore = Regex::new(r"\{\{_}\}").unwrap();
         let text_re = Regex::new(r"[a-zA-Z0-9\._]+").unwrap();
-        // The dissamebler alternates between upper- and lowercase hex, making matching addresses
```

Comment:
Unrelated comment: the disassembler we use in tests is different from the one we use elsewhere (e.g. in PT). We used a different one because there was a sanitiser issue, which I believe is now fixed. We could probably use `iced` across the board if there is a hunger to minimise deps and maybe fix this casing issue.

Reply:
I have no idea. I definitely don't want to think about that in this PR :)

Reply:
Of course. Hence flagging it unrelated.